src/BlazorWebView/src/core/BlazorWebViewRegistrationExtensions.cs

```diff
@@ -1,17 +1,39 @@
 ﻿using System;
+using System.Reflection;
+using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Maui.Hosting;
 
 namespace Microsoft.AspNetCore.Components.WebView.Maui
 {
 	public static class BlazorWebViewRegistrationExtensions
 	{
 		public static TAppHostBuilder RegisterBlazorMauiWebView<TAppHostBuilder>(this TAppHostBuilder appHostBuilder) where TAppHostBuilder : IAppHostBuilder
+		{
+			var entryAssembly = Assembly.GetEntryAssembly();
+			if (entryAssembly == null)
```

Comment:
Just curious - how is there sometimes no entry assembly?

Reply:
A recursive application has no entry assembly, e.g. A --> B --> C --> A. Just kidding I have no idea, but the API says it can be null (in terms of annotations, anyway).

Reply:
Oh I guess it can happen when the entry assembly is unmanaged, which I suppose might sometimes be the case in MAUI. The samples all pass in an explicit assembly, and I plan for the project template to do the same.

Reply:
If we could rely on the entry assembly being correct, we wouldn't need `BlazorAssetsAssemblyConfiguration` at all because the webviews could just directly hook up their manifest file provider to `Assembly.GetEntryAssembly()`. I guess there must be scenarios where that's not adequate.

Presumably this can all go away anyway once we have static assets wired up to MAUI assets and not embedded in the assembly, right?

Reply:
For now the entry assembly isn't the right assembly in some key scenarios: if you're using multi-project scenarios then the entry assembly will either be null, or will be the wrong assembly (e.g. MyApp.WinUI.dll), whereas we need MyApp.dll.

But, yes, with MauiAssets, this wouldn't be needed at all, because MauiAssets itself will always Do The Needful™.